pghoard/basebackup.py

```diff
@@ -412,20 +410,123 @@ def add_entry(archive_path, local_path, *, missing_ok):
+            # pylint: disable=bad-continuation
+            with rohmufile.file_writer(
+                    compression_algorithm=self.config["compression"]["algorithm"],
+                    compression_level=self.config["compression"]["level"],
+                    rsa_public_key=rsa_public_key,
+                    fileobj=raw_output_obj) as output_obj:
+                with tarfile.TarFile(fileobj=output_obj, mode="w") as output_tar:
+                    self.write_files_to_tar(files=files_to_backup, tar=output_tar)
+
+                input_size = output_obj.tell()
+
+            result_size = raw_output_obj.tell()
+            # Make the file persist over the with-block with this hardlink
+            os.link(raw_output_obj.name, chunk_path)
+
+        rohmufile.log_compression_result(
+            encrypted=True if encryption_key_id else False,
+            elapsed=time.monotonic() - start_time,
+            original_size=input_size,
+            result_size=result_size,
+            source_name="$PGDATA files ({})".format(len(files_to_backup)),
+            log_func=self.log.info,
+        )
+
+        metadata = {
+            "compression-algorithm": self.config["compression"]["algorithm"],
+            "encryption-key-id": encryption_key_id,
+            "format": "pghoard-bb-v2",
+            "original-file-size": input_size,
+        }
+        if extra_metadata:
+            metadata.update(extra_metadata)
+        self.transfer_queue.put({
+            "callback_queue": callback_queue,
+            "file_size": result_size,
+            "filetype": filetype,
+            "local_path": chunk_path,
+            "metadata": metadata,
+            "site": self.site,
+            "type": "UPLOAD",
+        })
+
+        # Get the name of the chunk and the name of the parent directory (ie backup "name")
+        chunk_name = "/".join(chunk_path.split("/")[-2:])
+        return chunk_name, input_size, result_size
+
+    def wait_for_chunk_transfer_to_complete(self, chunk_count, upload_results, chunk_callback_queue, start_time):
+        try:
+            upload_results.append(chunk_callback_queue.get(timeout=3.0))
+            self.log.info("Completed a chunk transfer successfully: %r", upload_results[-1])
+            return True
+        except Empty:
+            self.log.warning("Upload status: %r/%r handled, time taken: %r", len(upload_results), chunk_count,
+                             time.monotonic() - start_time)
+        return False
+
+    def create_and_upload_chunks(self, chunks, data_file_format, temp_base_dir):
+        start_time = time.monotonic()
+        chunk_files = []
+        upload_results = []
+        chunk_callback_queue = Queue()
+        chunks_on_disk = 0
+        i = 0
+
+        while i < len(chunks):
+            if chunks_on_disk < self.config["backup_sites"][self.site].get("basebackup_chunk_count", 5):
```

Comment:
We've been trying to set the default values in `config.py`, please do so for this value, too.